### PR TITLE
Send labels with the initial ws connection headers

### LINKED ISSUE #1136: Master does not take account node engine labels on initial agent connection
Master sends node information back to agent before agent  sends docker engine labels to master.

This is troublesome in situations where docker engine labels have a custom `region` label because the master will resolve overlay (weave) peer addresses using this label information. Basically this means that node tries to connect to other nodes via public ip which in most cases is not a valid assumption.

### PR DESCRIPTION
**Pull request by jnummelin**
fixes #1136

### CODE REVIEW

**Comment by jakolehm** [comment]
Not 100% sure this is the right way... should we send node information on websocket connect event (now it's `websocket:connected` which happens too late)?

**Comment by jakolehm** [comment]
As discussed, sending info on websocket connect does not help here because it's also too late.

**Comment by jnummelin** [comment]
If we don't want to use the headers to send the info, I can think of two alternative approaches. Both need some refactoring on the node plugging sequence.

sequencediagram1

In this case server would request the info from agent and after receiving it would execute the node plugging functionality. This would require new RPC api on agent side and refactoring of the node_plugger sequence.

sequencediagram2

In this case agent would send the full node info data after the ws connection is opened. Server executes the node plugger functionality after receiving this initial node info. This would require some mechanism to keep track of whether or not initial node plugging functionality is executed or not as agent periodically send the full info to the server.


Some other options, all which might be bad ideas :) :
- Send the node_info data on the initial connect as base64 encoded query param
- Send the node_info data on the initial connect as base64 encoded header field


As for now, the only critical extra info is the labels and to send those the custom header field is pretty simple solution.

File: agent/lib/kontena/websocket_client.rb [hunk 1]
```diff
@@ -161,6 +162,10 @@ def host_id
       Docker.info['ID']
     end
 
+    def labels
+      Docker.info['Labels'].join(',')
```
Comment (jakolehm) [inline agent/lib/kontena/websocket_client.rb]: Is this always an array? Maybe it would be safer to cast `.to_a` first?

**Comment by jakolehm** [comment]
Ok, maybe using headers is fine for bug fixing this race condition. We can refactor this logic later to match your diagram.

File: agent/lib/kontena/websocket_client.rb (hunk 1 above)
Comment (jnummelin) [inline agent/lib/kontena/websocket_client.rb]: Good point, will change.

**Pull request merged**